Ros2 control support package added

EXPX3 opened the pull request.

Commit:
created new package  'kuka_ros2_control_support'

Commit:
single ros2_control launch file for all of the kuka robots  except 'kuka_lbr_iiwa_support'

Commit:
moved common_macro.ros2_control.xacro to kuka_ros2_control_support

gavanderhoorn commented on kuka_kr10_support/urdf/kr10r1100sixx.xacro:
```diff
@@ -4,7 +4,7 @@
 
   <xacro:include filename="$(find kuka_kr10_support)/urdf/kr10r1100sixx_macro.xacro" />
-  <xacro:include filename="$(find kuka_resources)/urdf/common_macro.ros2_control.xacro"/>
+  <xacro:include filename="$(find kuka_ros2_control_support)/urdf/common_macro.ros2_control.xacro"/>
```
High-level comment: this appears to go in the direction discussed in #11, but I believe what @destogl meant was for none of the `.xacro`s to contain any reference to `ros2_control` any more.

So the only package which depends on -- and adds compatibility for -- `ros2_control` would be `kuka_ros2_control_support`.

All other packages have no `exec_depend`s, nor any references to `ros2_control` any more.

Concretely: this `xacro:include` should be removed (here and in all other files). All the `xacro:arg`s which are related to `ros2_control` should also be removed. And the macro call below should also be removed.

In essence: all top-level `.xacro`s (and the underlying `_macro.xacro`s) should be reverted to how they were. `ros2_control` support is added by wrapping the `_macro.xacro`s and combining them with the files from/in `kuka_ros2_control_support`.

destogl replied:
Yes, exactly. We should load the description macro files of robots into ros2_control related files in the `kuka_ros2_control_support` package.

This could though make a bit tricky dependencies, but we can consider this issue later. by explicitly adding this int the user manual.

Commit:
making base kuka files independent of ros2_control

Commit:
making base kuka files independent of ros2_control-2

Commit:
common xacro and ros2_controlxacro

Commit:
functional ros2_control support for kuka base models

Commit:
correction

Commit:
removed unused xacro

Commit:
removed unused xacro and added one useful

Commit:
removed arguments not necessary

Commit:
removed dependency not needed in ros2

destogl commented on kuka_ros2_control_support/urdf/common_kuka.xacro:
```diff
@@ -0,0 +1,86 @@
+
+  <xacro:arg name="prefix" default="" />
+  <xacro:arg name="use_mock_hardware" default="false" />
+  <xacro:arg name="robot_description_package" default="kuka_kr3_support"/>
+  <xacro:arg name="robot_description_macro_file" default="kr3r540_macro.xacro"/>
+  <xacro:arg name="robot_name" default="kuka_kr3r540"/>
```
```suggestion
  <xacro:arg name="robot_description_package"/>
  <xacro:arg name="robot_description_macro_file"/>
  <xacro:arg name="robot_name"/>
```

EXPX3 committed:
[URDF] Adding inertial and using `find` for getting package paths for simulators to work (#13)

Co-authored-by: Dr. Denis <denis@stoglrobotics.de>

Commit:
ran pre-commit for all ecept *_rsi_* pkgs

EXPX3 committed:
Update kuka_ros2_control_support/urdf/common_kuka.xacro

Co-authored-by: Dr. Denis <denis@stoglrobotics.de>

EXPX3 commented on kuka_ros2_control_support/urdf/common_macro.ros2_control.xacro:
```diff
@@ -28,7 +31,7 @@
           <param name="initial_value">0.0</param>
         </state_interface>
-	  </joint>
+	    </joint>
```
```suggestion
	   </joint>
```

EXPX3 committed:
Update kuka_ros2_control_support/urdf/common_macro.ros2_control.xacro

EXPX3 commented on kuka_ros2_control_support/urdf/common_macro.ros2_control.xacro:
```diff
@@ -28,7 +31,7 @@
           <param name="initial_value">0.0</param>
         </state_interface>
-	  </joint>
+	   </joint>
```
```suggestion
	  </joint>
```

EXPX3 commented on kuka_ros2_control_support/urdf/common_macro.ros2_control.xacro:
```diff
@@ -37,7 +40,7 @@
           <param name="initial_value">0.0</param>
         </state_interface>
-	  </joint>
+	    </joint>
```
```suggestion
	  </joint>
```

EXPX3 committed:
Apply suggestions from code review

Commit:
included kuka_lbr in the comman launch and xacro files

Commit:
ran-precommit

Commit:
kuka_ros2_control_support, complete macro  including gazebo plugins

Robotgir commented:
closing in favor of #15 
#15 includes both the packages

EXPX3 committed:
Merge branch 'rolling' into ros2_control_support_package_added

Commit:
corrections

Commit:
updated readme on how to use

EXPX3 committed:
Update README.md

muritane commented on README.md:
```diff
@@ -44,3 +44,111 @@ ROS2 Distro | Branch | Build status | Documentation | Released packages
+                "kuka_kr16_support",
+                "kuka_kr120_support",
+                "kuka_kr150_support",
+                "kuka_kr210_support",
+            ]
```
```suggestion
                "kuka_lbr_iiwa_support",
            ]
```

muritane replied:
lbr iiwa should be added

muritane replied:
```suggestion
                "kuka_lbr_iiwa_support",
            ]
```

muritane commented on README.md:
```diff
@@ -44,3 +44,111 @@ ROS2 Distro | Branch | Build status | Documentation | Released packages
+                "kr120r2500pro_macro.xacro",
+                "kr150_2_macro.xacro",
+                "kr150r3100_2_macro.xacro",
+                "kr210l150_macro.xacro",
+            ]
```
```suggestion
                "kr210l150_macro.xacro",
                "lbr_iiwa_14_r820_macro.xacro"
            ]
```

muritane replied:
lbr iiwa should be added

muritane replied:
```suggestion
                "lbr_iiwa_14_r820_macro.xacro",
            ]
```

muritane commented on README.md:
```diff
@@ -44,3 +44,111 @@ ROS2 Distro | Branch | Build status | Documentation | Released packages
+                "kuka_kr6_support",
+                "kuka_kr10_support",
+                "kuka_kr16_support",
+                "kuka_kr120_support",
+                "kuka_kr150_support",
+                "kuka_kr210_support",
+            ]
+    
+    "robot_description_macro_file",
+            choices=[
+                "kr3r540_macro.xacro",
+                "kr5_arc_macro.xacro",
+                "kr6r700sixx_macro.xacro",
+                "kr6r900_2_macro.xacro",
+                "kr6r900sixx_macro.xacro",
+                "kr10r900_2_macro.xacro",
+                "kr10r1100sixx_macro.xacro",
+                "kr10r1420_macro.xacro",
+                "kr16_2_macro.xacro",
+                "kr120r2500pro_macro.xacro",
+                "kr150_2_macro.xacro",
+                "kr150r3100_2_macro.xacro",
+                "kr210l150_macro.xacro",
+            ]
+    "robot_name",
+            choices=[
+                "kuka_kr3r540",
+                "kuka_kr5_arc",
+                "kuka_kr6r700sixx",
+                "kuka_kr6r900_2",
+                "kuka_kr6r900sixx",
+                "kuka_kr10r900_2",
+                "kuka_kr10r1100sixx",
+                "kuka_kr10r1420",
+                "kuka_kr16_2",
+                "kuka_kr120r2500pro",
+                "kuka_kr150_2",
+                "kuka_kr150r3100_2",
+                "kuka_kr210l150",
+            ]
```
```suggestion
    "robot_description_package",
            choices=[
                "kuka_kr3_support",
                "kuka_kr5_support",
                "kuka_kr6_support",
                "kuka_kr10_support",
                "kuka_kr16_support",
                "kuka_kr120_support",
                "kuka_kr150_support",
                "kuka_kr210_support",
                "kuka_lbr_iiwa_support"
            ]
    
    "robot_description_macro_file",
            choices=[
                "kr3r540_macro.xacro",
                "kr5_arc_macro.xacro",
                "kr6r700sixx_macro.xacro",
                "kr6r900_2_macro.xacro",
                "kr6r900sixx_macro.xacro",
                "kr10r900_2_macro.xacro",
                "kr10r1100sixx_macro.xacro",
                "kr10r1420_macro.xacro",
                "kr16_2_macro.xacro",
                "kr120r2500pro_macro.xacro",
                "kr150_2_macro.xacro",
                "kr150r3100_2_macro.xacro",
                "kr210l150_macro.xacro",
                "lbr_iiwa_14_r820_macro.xacro",
            ]
    "robot_name",
            choices=[
                "kuka_kr3r540",
                "kuka_kr5_arc",
                "kuka_kr6r700sixx",
                "kuka_kr6r900_2",
                "kuka_kr6r900sixx",
                "kuka_kr10r900_2",
                "kuka_kr10r1100sixx",
                "kuka_kr10r1420",
                "kuka_kr16_2",
                "kuka_kr120r2500pro",
                "kuka_kr150_2",
                "kuka_kr150r3100_2",
                "kuka_kr210l150",
                "kuka_lbr_iiwa_14_r820",
            ]
```

muritane replied:
```suggestion
                "kuka_lbr_iiwa_14_r820",
            ]
```

destogl committed:
Apply suggestions from code review

Co-authored-by: muritane <31107191+muritane@users.noreply.github.com>

destogl reviewed (approved):
Looking good!

destogl commented:
@muritane you created a script for testing this, didn't you?

Can we attach this?

muritane committed:
added test for kuka_ros2_control_support

muritane committed:
pre-commit changes for ros2_control_support_test

Commit:
added initial_positions_file arg

destogl commented on kuka_ros2_control_support/urdf/common_macro.ros2_control.xacro:
```diff
@@ -4,13 +4,14 @@
                robot_description_package"
                >
-
+    <xacro:property name="initial_positions" value="${load_yaml(initial_positions_file)['initial_positions']}"/>
```
we should also utilize this agument lower under `initial_value` tag

mamueluth committed:
add control node and ip address/port options

destogl committed:
Merge branch 'rolling' into ros2_control_support_package_added

mamueluth committed:
add joint trajecotry controller goals publisher

destogl commented on README.md:
````diff
@@ -44,3 +44,123 @@ ROS2 Distro | Branch | Build status | Documentation | Released packages
+    ros2 launch kuka_ros2_control_support test_ros2_control_kuka.launch.py robot_description_package:=kuka_kr16_support robot_description_macro_file:=kr16_2_macro.xacro robot_name:=kuka_kr16_2 controllers_file:=kuka_6dof_controllers.yaml
+
+    ```
+
````
```suggestion
```

destogl commented on kuka_experimental.rolling.repos:
```diff
@@ -1,6 +1,13 @@
-#     type: git
-#     url: git@github.com:<some_github_namespace>/<some_ros_package>.git
-#     version: master
+  kuka_experimental:
+    type: git
+    url: git@github.com:StoglRobotics-forks/kuka_experimental.git
+    version: rolling
```
```suggestion
```

destogl commented on kuka_rsi_hw_interface/CMakeLists.txt:
```diff
@@ -1,66 +1,83 @@
-cmake_minimum_required(VERSION 3.1.0)
+cmake_minimum_required(VERSION 3.5)
```
```suggestion
cmake_minimum_required(VERSION 3.16)
```

destogl committed:
Update first part of the readme file.

destogl committed:
Removing, cleaning, renaming and moving configuration files.

destogl committed:
Revert some file and optimize CMakeLists.txt

destogl committed:
Some cleanups.

destogl committed:
Cleaning.

destogl committed:
Working with Mock Hardware.

muritane committed:
ros2 control support test works

muritane committed:
added joint trajectory test

muritane committed:
Update kuka_ros2_control_support/test/ros2_control_support.py

Co-authored-by: gwalck <guillaume.walck@stoglrobotics.de>

muritane committed:
Update initial_positions_all_zeros.yaml

muritane committed:
add tests for joint and traj state retrieval, init_pos arg, start_rviz=False

muritane committed:
[test_bringup.launch.py] set use_mock_hardware default to true

gwalck committed:
Added a tcp transform test to validate RSP and TFs

Due to many robots with tip at various poses,
the test verifies only if the tool0 frame
can be resolved and is non-null

gwalck committed:
Moved tf_listener to func to avoid thread issues

With the listener within the main class,
strange wait_for_message issues arise

destogl committed:
Cleanup more files.

gwalck committed:
Added missing line to cleanly initialize executor
